crates/shielded_token/src/masp/shielded_wallet.rs

```diff
@@ -2007,21 +2245,37 @@ mod test_shielded_wallet {
                         MerklePath::from_path(vec![], 0),
                     )
                 );
-
                 let vk = arbitrary_vk();
                 let pa = arbitrary_pa();
                 let asset_data = AssetData {
                     token: incentivized_token.clone(),
                     denom: 0.into(),
                     position: MaspDigitPos::Zero,
-                    epoch: Some(MaspEpoch::new(1)),
+                    epoch: Some(MaspEpoch::new(2)),
                 };
 
                 wallet.add_note(
-                    create_note(asset_data.clone(), principal, pa),
+                    create_note(asset_data, principal, pa),
                     vk,
                 );
 
+                // If this flag is clear we test that the estimation logic can
+                // hanlde rewards for assets coming only from the current epoch,
+                // i.e. assets for which there are no conversions yet.
+                if shield_at_previous_epoch {
+                    let asset_data = AssetData {
+                        token: incentivized_token.clone(),
+                        denom: 0.into(),
+                        position: MaspDigitPos::Zero,
+                        epoch: Some(MaspEpoch::new(1)),
```

Comment:
isn't the current epoch 2?

Reply:
Yes but if `shielded_at_previous_epoch` is **clear** then we only shield at the current epoch (2), if it is set instead we also shield at the previous one, let me rephrase the comment